src/transforms/onehot.jl

```diff
@@ -3,34 +3,40 @@
 # ------------------------------------------------------------------
 
 """
-    OneHot(col)
-    
+    OneHot(col; categorical=true)
+
 Transforms categorical column `col` into one-hot columns of levels
 returned by the `levels` function of CategoricalArrays.jl.
+The `categorical` option can be used to convert resulting
+columns to categorical arrays as opposed to boolean vectors.
 
 # Examples
 
 ```julia
 OneHot(1)
 OneHot(:a)
 OneHot("a")
+OneHot("a", categorical=false)
 ```
 """
 struct OneHot{S<:ColSpec} <: StatelessFeatureTransform
   colspec::S
-  function OneHot(col::Col)
+  categorical::Bool
```

Comment:
I wonder if we should abbreviate the field to something short like `categ`. We do similar abbreviations in `Quantile(dist=Normal())` for example to avoid typing `distribution=Normal()`.